TESTS/mbed_hal/flash/functional_tests/main.cpp

```diff
@@ -74,13 +74,13 @@ static void delay_loop(uint32_t count)
     : "cc"
   );
 }
-#else // GCC
+#elif  defined ( __GNUC__ ) ||  (defined(__ARMCC_VERSION) && (__ARMCC_VERSION >= 6010050))
```

Comment:
It looks like this will work but I don't usually like #elif without a corresponding default #else case.  I think we can live with it for now though